File: src/main/java/bilibili/lingaly/tutorial/block/ModBlocks.java
package bilibili.lingaly.tutorial.block;

import bilibili.lingaly.tutorial.ModItemGroups;
import bilibili.lingaly.tutorial.Tutorial;
import bilibili.lingaly.tutorial.block.advanced.WeakBlock;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.fabricmc.fabric.api.object.builder.v1.block.FabricBlockSettings;
import net.minecraft.block.Block;
import net.minecraft.block.Material;
import net.minecraft.block.SlabBlock;
import net.minecraft.block.StairsBlock;
import net.minecraft.item.BlockItem;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModBlocks {
    public static Block TUTORIAL_BLOCK = registerBlock("tutorial_block",
            new Block(FabricBlockSettings.of(Material.METAL).strength(5f).requiresTool()),
            ModItemGroups.TUTORIAL);
    public static Block TUTORIAL_ORE = registerBlock("tutorial_ore",
            new Block(FabricBlockSettings.of(Material.STONE).strength(4f).requiresTool()),
            ModItemGroups.TUTORIAL);
    public static Block DEEPSLATE_TUTORIAL_ORE = registerBlock("deepslate_tutorial_ore",
            new Block(FabricBlockSettings.of(Material.STONE).strength(6f).requiresTool()),
            ModItemGroups.TUTORIAL);
    public static Block RAW_TUTORIAL_BLOCK = registerBlock("raw_tutorial_block",
            new Block(FabricBlockSettings.of(Material.METAL).strength(5f).requiresTool()),
            ModItemGroups.TUTORIAL);

    public static Block WEAK_BLOCK= registerBlock("tutorial_block",
            new WeakBlock(FabricBlockSettings.of(Material.METAL).strength(1f)),
            ModItemGroups.TUTORIAL);

    public static Block TUTORIAL_STAIR = registerBlock("tutorial_stair",
            new StairsBlock(ModBlocks.TUTORIAL_BLOCK.getDefaultState(),
                    FabricBlockSettings.of(Material.METAL)
                    .strength(4f)),
            ModItemGroups.TUTORIAL);

    public static Block TUTORIAL_SLAB = registerBlock("tutorial_slab",
            new SlabBlock(FabricBlockSettings.of(Material.METAL)
                            .strength(4f)),
            ModItemGroups.TUTORIAL);

    private static Block registerBlock(String name, Block block, ItemGroup itemGroup){
        registerBlockItem(name, block, itemGroup);
        return Registry.register(Registry.BLOCK, new Identifier(Tutorial.MOD_ID, name), block);
    }

    private static Item registerBlockItem(String name, Block block, ItemGroup itemGroup){
        return Registry.register(Registry.ITEM, new Identifier(Tutorial.MOD_ID, name),
                new BlockItem(block, new FabricItemSettings().group(itemGroup)));
    }

    public static void ModBlockRegister(){}

}

File: src/main/kotlin/bilibili/lingaly/ktmccouse/block/ModBlocks.kt
package bilibili.lingaly.ktmccouse.block

import bilibili.lingaly.easyfabricregisterlib.EFRLModRegister
import bilibili.lingaly.ktmccouse.Couse
import bilibili.lingaly.ktmccouse.ModItemGroups
import bilibili.lingaly.ktmccouse.block.advanced.Weakblock
import net.fabricmc.fabric.api.`object`.builder.v1.block.FabricBlockSettings
import net.minecraft.block.Material
import net.minecraft.block.SlabBlock
import net.minecraft.block.StairsBlock

object ModBlocks {

    private const val MOD_ID = Couse.MOD_ID

    val TIN_BLOCK = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "tin_block",
            Material
                .METAL,
            5f,
            true,
            ModItemGroups.TIN
        )

    @Suppress("unused")
    val TIN_ORE = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "tin_ore",
            Material
                .STONE,
            4f,
            true,
            ModItemGroups.TIN
        )

    @Suppress("unused")
    val DEEPSLATE_TIN_ORE = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "deepslate_tin_ore",
            Material
                .STONE,
            6f,
            true,
            ModItemGroups
                .TIN
        )

    @Suppress("unused")
    val RAW_TIN_BLOCK = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "raw_tin_block",
            Material
                .METAL,
            5f,
            true,
            ModItemGroups.TIN
        )

    @Suppress("unused")
    val WEAK_BLOCK = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "weak_block",
            Weakblock(
                FabricBlockSettings
                    .of(
                        Material
                            .METAL
                    )
                    .strength(
                        5f
                    )
            ),
            ModItemGroups
                .TIN
        )

    @Suppress("unused")
    val TIN_STAIR = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "tin_stair",
            StairsBlock(
                TIN_BLOCK
                    .defaultState,
                FabricBlockSettings
                    .of(
                        Material
                            .METAL
                    )
                    .strength(
                        5f
                    )
                ),
            ModItemGroups.TIN
        )

    @Suppress("unused")
    val TIN_SLAB = EFRLModRegister
        .registerBlock(
            MOD_ID,
            "tin_slab",
            SlabBlock(
                FabricBlockSettings
                    .of(
                        Material
                            .METAL
                    )
                    .strength(
                        5f
                    )
            ),
            ModItemGroups
                .TIN
        )

    fun modBlockRegister() {}

}